s3/init_test.go

```diff
@@ -12,15 +15,44 @@ const (
 
 var testStore *Storage
 
+const (
+	// minioImage is the default image used for running S3 in tests.
+	minioImage              = "docker.io/minio/minio:RELEASE.2024-08-17T01-24-54Z"
+	minioImageEnvVar string = "TEST_MINIO_IMAGE"
+	minioUser        string = "minio-user"
+	minioPass        string = "minio-password"
+)
+
 func TestMain(m *testing.M) {
+	img := minioImage
+	if imgFromEnv := os.Getenv(minioImageEnvVar); imgFromEnv != "" {
+		img = imgFromEnv
+	}
+
+	ctx := context.Background()
+
+	c, err := minio.Run(ctx,
+		img,
+		minio.WithUsername(minioUser),
+		minio.WithPassword(minioPass),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	conn, err := c.ConnectionString(ctx)
+	if err != nil {
+		panic(err)
+	}
+
 	testStore = New(
 		Config{
 			Bucket:   bucket,
-			Endpoint: "http://127.0.0.1:9000/",
+			Endpoint: "http://" + conn,
```

Comment:
Something is not happy with this

Reply:
I'm debugging this atm, will post updates as commits, sorry about that

Reply:
It seems and endpoint resolver is missing the last S3 updates. IIRC I had to do the same for the S3 examples with localstack: https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/configure-endpoints.html#with-both

This is needed to avoid the BaseEndpoint to prepend the bucket name, as shown in the test errors